File: src/api/types/events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
pub struct SecurityEventsResponse {
    pub(crate) total: u64,
    pub(crate) limit: u32,
    pub(crate) offset: u32,
    pub(crate) events: Vec<SecurityEventResponse>,
}

#[derive(Debug, Serialize)]
pub struct BehaviorProfilesResponse {
    pub(crate) total: u64,
    pub(crate) profiles: Vec<BehaviorProfileResponse>,
}

#[derive(Debug, Serialize)]
pub struct BehaviorProfileResponse {
    pub(crate) identity: String,
    pub(crate) latest_seen_at: i64,
    pub(crate) score: u32,
    pub(crate) dominant_route: Option<String>,
    pub(crate) focused_document_route: Option<String>,
    pub(crate) distinct_routes: usize,
    pub(crate) repeated_ratio: u32,
    pub(crate) document_repeated_ratio: u32,
    pub(crate) interval_jitter_ms: Option<u64>,
    pub(crate) document_requests: usize,
    pub(crate) non_document_requests: usize,
    pub(crate) challenge_count_window: usize,
    pub(crate) session_span_secs: u64,
    pub(crate) flags: Vec<String>,
    pub(crate) latest_route: String,
    pub(crate) latest_kind: String,
}

#[derive(Debug, Serialize)]
pub struct SecurityEventResponse {
    pub(crate) id: i64,
    pub(crate) layer: String,
    pub(crate) provider: Option<String>,
    pub(crate) provider_event_id: Option<String>,
    pub(crate) provider_site_id: Option<String>,
    pub(crate) provider_site_name: Option<String>,
    pub(crate) provider_site_domain: Option<String>,
    pub(crate) action: String,
    pub(crate) reason: String,
    pub(crate) details_json: Option<String>,
    pub(crate) source_ip: String,
    pub(crate) dest_ip: String,
    pub(crate) source_port: i64,
    pub(crate) dest_port: i64,
    pub(crate) protocol: String,
    pub(crate) http_method: Option<String>,
    pub(crate) uri: Option<String>,
    pub(crate) http_version: Option<String>,
    pub(crate) created_at: i64,
    pub(crate) handled: bool,
    pub(crate) handled_at: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct EventUpdateRequest {
    pub(crate) handled: bool,
}

#[derive(Debug, Serialize)]
pub struct BlockedIpsResponse {
    pub(crate) total: u64,
    pub(crate) limit: u32,
    pub(crate) offset: u32,
    pub(crate) blocked_ips: Vec<BlockedIpResponse>,
}

#[derive(Debug, Serialize)]
pub struct BlockedIpResponse {
    pub(crate) id: i64,
    pub(crate) provider: Option<String>,
    pub(crate) provider_remote_id: Option<String>,
    pub(crate) ip: String,
    pub(crate) reason: String,
    pub(crate) blocked_at: i64,
    pub(crate) expires_at: i64,
}

#[derive(Debug, Deserialize)]
pub struct BlockedIpCreateRequest {
    pub(crate) ip: String,
    pub(crate) reason: String,
    pub(crate) duration_secs: Option<u64>,
}

#[derive(Debug, Deserialize)]
pub struct BlockedIpsBatchUnblockRequest {
    pub(crate) ids: Vec<i64>,
}

#[derive(Debug, Serialize)]
pub struct BlockedIpsBatchUnblockResponse {
    pub(crate) success: bool,
    pub(crate) requested: u32,
    pub(crate) unblocked: u32,
    pub(crate) failed: u32,
    pub(crate) failed_ids: Vec<i64>,
    pub(crate) message: String,
}

#[derive(Debug, Deserialize, Default)]
pub struct BlockedIpsCleanupExpiredRequest {
    pub(crate) source_scope: Option<String>,
    pub(crate) provider: Option<String>,
    pub(crate) blocked_from: Option<i64>,
    pub(crate) blocked_to: Option<i64>,
    pub(crate) expires_before: Option<i64>,
}

#[derive(Debug, Serialize)]
pub struct BlockedIpsCleanupExpiredResponse {
    pub(crate) success: bool,
    pub(crate) cleaned: u32,
    pub(crate) runtime_unblocked: u32,
    pub(crate) message: String,
}

#[derive(Debug, Deserialize, Default)]
pub struct EventsQueryParams {
    pub(crate) limit: Option<u32>,
    pub(crate) offset: Option<u32>,
    pub(crate) layer: Option<String>,
    pub(crate) provider: Option<String>,
    pub(crate) provider_site_id: Option<String>,
    pub(crate) source_ip: Option<String>,
    pub(crate) action: Option<String>,
    pub(crate) blocked_only: Option<bool>,
    pub(crate) handled_only: Option<bool>,
    pub(crate) created_from: Option<i64>,
    pub(crate) created_to: Option<i64>,
    pub(crate) sort_by: Option<String>,
    pub(crate) sort_direction: Option<String>,
}

#[derive(Debug, Deserialize, Default)]
pub struct BlockedIpsQueryParams {
    pub(crate) limit: Option<u32>,
    pub(crate) offset: Option<u32>,
    pub(crate) source_scope: Option<String>,
    pub(crate) provider: Option<String>,
    pub(crate) ip: Option<String>,
    pub(crate) keyword: Option<String>,
    pub(crate) active_only: Option<bool>,
    pub(crate) blocked_from: Option<i64>,
    pub(crate) blocked_to: Option<i64>,
    pub(crate) sort_by: Option<String>,
    pub(crate) sort_direction: Option<String>,
}

File: dome-vue/src/features/events/types/events.ts
import type { ApiQueryValue } from '@/shared/types/common'

export interface SecurityEventItem {
  id: number
  layer: string
  provider: string | null
  provider_event_id: string | null
  provider_site_id: string | null
  provider_site_name: string | null
  provider_site_domain: string | null
  action: string
  reason: string
  details_json: string | null
  source_ip: string
  dest_ip: string
  source_port: number
  dest_port: number
  protocol: string
  http_method: string | null
  uri: string | null
  http_version: string | null
  created_at: number
  handled: boolean
  handled_at: number | null
  decision_summary: SecurityEventDecisionSummary | null
}

export interface SecurityEventDecisionSummary {
  primary_signal: string
  identity_state: string | null
  client_ip_source: string | null
  forward_header_valid: boolean | null
  l4_overload_level: string | null
  l7_rule_inspection_mode: string | null
  cc_action: string | null
  behavior_action: string | null
  labels: string[]
}

export interface StoragePressureSummaryDetails {
  mode: string
  action: string | null
  original_reason: string | null
  count: number
  source_scope: string | null
  route: string | null
  time_window_start: number | null
  time_window_end: number | null
  first_created_at: number | null
  last_created_at: number | null
}

export interface SecurityEventsResponse {
  total: number
  limit: number
  offset: number
  events: SecurityEventItem[]
}

export interface SecurityEventsSummaryResponse {
  total_events: number
  sampled_events: number
  generated_at: number
  by_action: SecurityEventCountItem[]
  by_layer: SecurityEventCountItem[]
  by_provider: SecurityEventCountItem[]
  by_primary_signal: SecurityEventCountItem[]
  top_source_ips: SecurityEventCountItem[]
  top_routes: SecurityEventCountItem[]
  top_reasons: SecurityEventCountItem[]
  hourly: SecurityEventHourlyItem[]
  aggregated: SecurityEventsAggregatedSummary
  representative_events: SecurityEventItem[]
}

export interface SecurityEventCountItem {
  key: string
  count: number
}

export interface SecurityEventHourlyItem {
  bucket_start: number
  count: number
  blocked: number
  alerted: number
  challenged: number
  dropped: number
}

export interface SecurityEventsAggregatedSummary {
  summary_events: number
  represented_events: number
  hotspot_events: number
  long_tail_events: number
}

export interface BehaviorProfileItem {
  identity: string
  source_ip: string | null
  latest_seen_at: number
  score: number
  dominant_route: string | null
  focused_document_route: string | null
  focused_api_route: string | null
  distinct_routes: number
  repeated_ratio: number
  document_repeated_ratio: number
  api_repeated_ratio: number
  interval_jitter_ms: number | null
  document_requests: number
  api_requests: number
  non_document_requests: number
  challenge_count_window: number
  session_span_secs: number
  flags: string[]
  latest_route: string
  latest_kind: string
  blocked: boolean
  blocked_at: number | null
  blocked_expires_at: number | null
  blocked_reason: string | null
}

export interface BehaviorProfilesResponse {
  total: number
  profiles: BehaviorProfileItem[]
}

export interface FingerprintProfileItem {
  identity: string
  identity_kind: string
  source_ip: string | null
  first_seen_at: number
  last_seen_at: number
  first_site_domain: string | null
  last_site_domain: string | null
  first_user_agent: string | null
  last_user_agent: string | null
  total_security_events: number
  total_behavior_events: number
  total_challenges: number
  total_blocks: number
  latest_score: number | null
  max_score: number
  latest_action: string | null
  reputation_score: number
  notes: string
}

export interface FingerprintProfilesResponse {
  total: number
  profiles: FingerprintProfileItem[]
}

export interface BehaviorSessionItem {
  session_key: string
  identity: string
  source_ip: string | null
  site_domain: string | null
  opened_at: number
  last_seen_at: number
  event_count: number
  challenge_count: number
  block_count: number
  latest_action: string | null
  latest_uri: string | null
  latest_reason: string | null
  dominant_route: string | null
  focused_document_route: string | null
  focused_api_route: string | null
  distinct_routes: number
  repeated_ratio: number
  document_repeated_ratio: number
  api_repeated_ratio: number
  document_requests: number
  api_requests: number
  non_document_requests: number
  interval_jitter_ms: number | null
  session_span_secs: number
  flags: string[]
}

export interface BehaviorSessionsResponse {
  total: number
  sessions: BehaviorSessionItem[]
}

export interface EventsQuery extends Record<string, ApiQueryValue> {
  limit?: number
  offset?: number
  layer?: string
  provider?: string
  provider_site_id?: string
  source_ip?: string
  action?: string
  identity_state?: string
  primary_signal?: string
  labels?: string
  blocked_only?: boolean
  handled_only?: boolean
  created_from?: number
  created_to?: number
  sort_by?: string
  sort_direction?: 'asc' | 'desc'
}

export interface BlockedIpItem {
  id: number
  provider: string | null
  provider_remote_id: string | null
  ip: string
  reason: string
  blocked_at: number
  expires_at: number
}

export interface BlockedIpsResponse {
  total: number
  limit: number
  offset: number
  blocked_ips: BlockedIpItem[]
}

export interface BlockedIpsQuery extends Record<string, ApiQueryValue> {
  limit?: number
  offset?: number
  source_scope?: 'all' | 'local' | 'remote'
  provider?: string
  ip?: string
  keyword?: string
  active_only?: boolean
  blocked_from?: number
  blocked_to?: number
  sort_by?: string
  sort_direction?: 'asc' | 'desc'
}

export interface BlockedIpCreatePayload {
  ip: string
  reason: string
  duration_secs?: number
}

export interface BlockedIpsBatchUnblockPayload {
  ids: number[]
}

export interface BlockedIpsBatchUnblockResponse {
  success: boolean
  requested: number
  unblocked: number
  failed: number
  failed_ids: number[]
  message: string
}

export interface BlockedIpsCleanupExpiredPayload {
  source_scope?: 'all' | 'local' | 'remote'
  provider?: string
  blocked_from?: number
  blocked_to?: number
  expires_before?: number
}

export interface BlockedIpsCleanupExpiredResponse {
  success: boolean
  cleaned: number
  runtime_unblocked: number
  message: string
}
